plugins/modules/dcnm_vrf.py

```diff
@@ -809,16 +907,32 @@ def diff_for_attach_deploy(self, want_a, have_a, replace=False):
                         break
 
             if not found:
-                if bool(want["isAttached"]):
-                    del want["isAttached"]
-                    want["deployment"] = True
-                    attach_list.append(want)
-                    if bool(want["is_deploy"]):
-                        dep_vrf = True
+                try:
+                    if want["isAttached"]:
+                        del want["isAttached"]
+                        want["deployment"] = True
+                        attach_list.append(want)
+                        if want["is_deploy"]:
```

Comment:
Same question here.  Should you be using `sefl.to_bool()`?

Reply:
Yep!  Will change.

Reply:
Changed these to the following.  Removed the try/except block since `to_bool()` will call `fail_json()` if it cannot convert any values.

```python
            if not found:
                if self.to_bool("isAttached", want):
                    del want["isAttached"]
                    want["deployment"] = True
                    attach_list.append(want)
                    if self.to_bool("is_deploy", want):
                        deploy_vrf = True
```